### PR TITLE
Close summarize window

### LINKED ISSUE #67: FEATURE: Close Kagi summarieser popup on <ESC>, and/or add "Close" button
It would be really nice, if the Kagi summarieser popup would close when `<ESC>` is pressed, or add a close button
If both things could be added, it would be really nice.

### PR DESCRIPTION
**Pull request by alabhyajindal**
Fixes #67

Added event listener for the `Escape` key and a close element. I'm unsure how to position the element.

Here's how it looks now, when time saved element is present:

image

When time saved element is not present:

image

### CODE REVIEW

**Comment by eyJhb** [comment]
I'm not a designer at all, but my initial thought was to make it like this (side-by-side).

image

**Review by BrunoBernardino — CHANGES_REQUESTED** [review]
I didn't pull this down or test yet, but the logic looks good. I would prefer it to be a button where you placed it (being side by side as @eyJhb suggested could easily lead to accidental clicks or confusion), but similar to the "Copy summary" button (with the text "Close"), just with a slightly smaller font and gray background.

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**
```
[the remark this answers is not in the document]
Comment (alabhyajindal) [reply .prettierignore]: Yes, sorry! I added this to disable my local Prettier settings and use `npm run format` to format the code. I have removed it now. 

Slight aside: I don't think `npm run format` is working. I get the output that it ran and formatted the code, but it doesn't.

File: shared/src/summarize_result.html [hunk 2]
```diff
@@ -57,6 +57,12 @@
       <div id="summary_stats" class="stats" style="display: none">
         <p><strong>Time saved:</strong> <span id="summary_stats_time_saved">0 minutes</span>.</p>
       </div>
+      <span id='close_summary' style='display: none;'>
+        <svg aria-hidden="true" height="100%" width="100%" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+          <path d="M15.9682 7.50003L7.5 15.9683L8.31882 16.7871L16.787 8.31885L15.9682 7.50003Z" fill="currentColor"/>
+          <path d="M8.31882 7.50003L7.5 8.31885L15.9682 16.7871L16.787 15.9683L8.31882 7.50003Z" fill="currentColor"/>
+        </svg>
+    </span>
```
[the remark this answers is not in the document]
Comment (alabhyajindal) [reply shared/src/summarize_result.html]: I made the change. I used the Eggshell color from the [Assets](https://kagi.com/assets) page.

File: shared/src/summarize_result.js [hunk 3]
```diff
@@ -66,6 +66,11 @@ async function setup() {
   }
 
   summaryStatsTimeSavedElement.innerText = '0 minutes';
+  const summaryCloseElement = document.getElementById('close_summary');
+  if (!summaryCloseElement) {
+    console.error('Could not find summarize close element');
+    return;
+  }
```
Comment (BrunoBernardino) [inline shared/src/summarize_result.js:73]: Can you please add a `summaryCloseElement.style.display = '';` after this? Similar to `summaryStatsElement`, it'll ensure that if the state is initialized in some other way in the future, it'll also reset the button's initial state.

**Review by BrunoBernardino — CHANGES_REQUESTED** [review]
This is looking great, thank you so much! I have a tiny request and this should be good to go!

File: .prettierignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrunoBernardino) [reply .prettierignore]: @alabhyajindal `npm run format` is working for me, is it possible you've got it disabled somehow, or your IDE is showing a stale version? (i.e. if you do `git diff` do you see changes?)

File: shared/src/summarize_result.js (hunk 3 above)
Comment (alabhyajindal) [reply shared/src/summarize_result.js:73]: Do you mean `summaryCloseElement.style.display = 'none';`? That's how the `summaryStatsElement` is handled.

File: .prettierignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (alabhyajindal) [reply .prettierignore]: Thanks. It looks like an IDE problem. I just opened the project with Vim and `npm run format` works as expected.

File: shared/src/summarize_result.js (hunk 3 above)
Comment (BrunoBernardino) [reply shared/src/summarize_result.js:74]: Exactly!

**Review by BrunoBernardino — APPROVED** [review]
Thank you so much for this!

**Comment by BrunoBernardino** [comment]
@vprelovac I'll cut a new release v0.6.1 later today with this.

**Pull request merged**